tools/stage_time_analysis.groovy

```diff
@@ -27,52 +27,62 @@ import groovy.json.JsonSlurper
           def json = new JsonSlurper().parseText(workflow)
           json.stages.each { stage ->
             if (stage.name.equalsIgnoreCase(findStage)) {
-              def duration = stage.durationMillis/MILLIS_IN_MINUTE
-              duration = duration.intValue()
-              println("  ==> Job: ${jobName} Stage: ${stage.name} durationMinutes: ${duration}")
-              sh("echo \'${jobName}:${duration}\' >> ${statsFile}")
+              def startTime = stage.startTimeMillis
+              // Did the job stage start after since?
+              if (startTime > since) {
+                def duration = stage.durationMillis/MILLIS_IN_MINUTE
+                duration = duration.intValue()
+                println("  ==> Job: ${jobName} Stage: ${stage.name} durationMinutes: ${duration}")
+                sh("echo \'${jobName}:${duration}\' >> ${statsFile}")
+              }
             }
           }
         }
   }
 
   // Iterate over all the jobs and get the duration stats
-  def getJobs(String statsFile, String findStage, String folder, Object listJobs) {
+  def getJobs(Long since, String statsFile, String findStage, String folder, Object listJobs) {
     listJobs.each { job ->
       if (job.jobs && job.jobs.size() > 0) {
         if (folder == null || folder.equals("")) {
-          getJobs(statsFile, findStage, "job/${job.name}", job.jobs)
+          getJobs(since, statsFile, findStage, "job/${job.name}", job.jobs)
         } else {
-          getJobs(statsFile, findStage, "${folder}/job/${job.name}", job.jobs)
+          getJobs(since, statsFile, findStage, "${folder}/job/${job.name}", job.jobs)
         }
       } else {
         def jobName="${folder}/job/${job.name}"
         if (folder == null || folder.equals("")) {
           jobName="job/${job.name}"
         }
 
-        getJobStageDuration(statsFile, jobName, findStage)
+        getJobStageDuration(since, statsFile, jobName, findStage)
       }
     }
   }
 
 node ("master") {
   def jenkinsUrl = "${params.JENKINS_URL}"
   def findStage = "${params.STAGE}"
+  def periodDays = params.PERIOD_DAYS as Long
```

Comment:
Is this ok not having a null catcher?

Reply:
These are mandatory params and have config defaults